nau_openedx_extensions/filters/pipeline.py

```diff
@@ -0,0 +1,38 @@
+"""
+Defined filters.
+"""
+
+from openedx_filters import PipelineStep
+from openedx_filters.learning.filters import CourseEnrollmentStarted
+
+from nau_openedx_extensions.filters.models import DomainsAllowedPerCourse
+
+
+class FilterEnrollmentByDomain(PipelineStep):   # pylint: disable=too-few-public-methods
+    """
+    Stop enrollment process raising PreventEnrollment exception if the user has an email
+    with a domain that is not in those allowed by a course in DomainsAllowedPerCourse.
+    Example usage:
+    Add the following configurations to your configuration file:
+        "OPEN_EDX_FILTERS_CONFIG": {
+            "org.openedx.learning.course.enrollment.started.v1": {
+                "fail_silently": false,
+                "pipeline": [
+                    "nau_openedx_extensions.filters.pipeline.FilterEnrollmentByDomain"
+                ]
+            }
+        }
+    """
+
+    def run_filter(self, user, course_key, mode):   # pylint: disable=unused-argument, no-self-use
+        """Filter."""
+
+        try:
+            course_and_domains = DomainsAllowedPerCourse.objects.get(course_key=course_key)
+            user_domain = user.email.split("@")[1]
+            if user_domain not in course_and_domains.get_allowed_domains_list:
+                raise CourseEnrollmentStarted.PreventEnrollment("You can't enroll on this course.")
```

Comment:
The message could be more specific.

"You can't enroll on this course because your email domain is not allowed. If you think this is an error, contact the course support"

Reply:
We also need that message to be translated.

Reply:
Here I am going to note down the use of wildcards: https://docs.python.org/3/library/fnmatch.html

And regarding having a more descriptive message, that message is printed on the server and not on the user side, I have to see how to put it on the user side, but I don't think it will be a problem.

Reply:
The UI shows the message from the server, which is relied by the http request. We do need to make the message descriptive. Translation is not an issue, we put it in the catalog and done.